components/windows_recall/windows_recall.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/windows_recall/windows_recall.h"
+
+#include <memory>
+
+#include "base/supports_user_data.h"
+#include "base/win/windows_version.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+
+namespace {
+
+struct WindowsRecallData : public base::SupportsUserData::Data {
+  static const int kKey = 0;  // Only address is used.
+
+  explicit WindowsRecallData(const bool enabled)
+      : is_windows_recall_enabled(enabled) {}
+
+  const bool is_windows_recall_enabled = false;
+};
+
+}  // namespace
+
+namespace windows_recall {
+
+bool IsWindowsRecallAvailable() {
+  return base::win::GetVersion() >= base::win::Version::WIN11;
+}
+
+void RegisterProfilePrefs(PrefRegistrySimple* registry) {
+  if (!IsWindowsRecallAvailable()) {
+    return;
+  }
+
+  registry->RegisterBooleanPref(prefs::kBlockWindowsRecall, true);
+}
+
+WindowsRecallState GetWindowsRecallState(
+    content::BrowserContext* browser_context) {
+  if (!IsWindowsRecallAvailable() || browser_context->IsOffTheRecord()) {
```

Comment:
why do we need this check for OTR? This is already handled by the chromium code when the original method returns false https://github.com/brave/brave-core/pull/29251/files